unit-test/background/toggle-reports.js

```diff
@@ -0,0 +1,329 @@
+import browser from 'webextension-polyfill'
+import load from '../../shared/js/background/load'
+import settings from '../../shared/js/background/settings'
+import ToggleReports from '../../shared/js/background/components/toggle-reports'
+import { _formatPixelRequestForTesting } from '../../shared/js/shared-utils/pixels'
+import tabManager from '../../shared/js/background/tab-manager'
+import tdsStorageStub from '../helpers/tds'
+
+describe('ToggleReports', () => {
+    const actualSentReports = []
+    let currentTabDetails = null
+    let currentTimestamp = 1
+    let settingsStorage = null
+    const toggleReports = new ToggleReports()
+    let toggleReportsConfig = null
+
+    // Dummy timestamps.
+    const oneDay = 86400000
+    const today = 1727101418823
+    const tomorrow = today + oneDay
+    const twoDaysTime = today + (oneDay * 2)
+    const threeDaysTime = today + (oneDay * 3)
+    const fourDaysTime = today + (oneDay * 4)
+    const fiveDaysTime = today + (oneDay * 5)
+    const sixDaysTime = today + (oneDay * 6)
+
+    beforeAll(async () => {
+        // Ensure the test configuration is being used, and keep a reference to
+        // the toggleReports feature configuration.
+        const { tdsData } = tdsStorageStub.stub()
+        toggleReportsConfig = tdsData.config.features.toggleReports
+
+        // Stub settings.
+        settingsStorage = new Map()
+        spyOn(settings, 'ready').and.returnValue(Promise.resolve())
+        spyOn(settings, 'getSetting').and.callFake(
+            name => settingsStorage.get(name)
+        )
+        spyOn(settings, 'updateSetting').and.callFake(
+            (name, value) => {
+                settingsStorage.set(name, value)
+            }
+        )
```

Comment:
I think properly injecting `settings`, `tdsStorage` and `tabManager` into the `ToggleReports`/`BreakageReports` components would eliminate the need for all these spies.

Reply:
Even if broken-site-report is also a componenent, `shouldDisplay` still needs to be static (see above) and therefore so does `clearExpiredResponses` and `countResponses`. So we would still need to import the `tabManager` and `settings` in the toggle-reports component. Therefore, we would still need these spies.